magento2-nginx/fpm-7.0/usr/local/share/magento2/install_magento_finalise.sh

```diff
@@ -9,14 +9,25 @@ else
 fi ;
 
 source /usr/local/share/bootstrap/common_functions.sh
-bin/magento setup:upgrade
+
+cd /app || exit 1
+
+# Preserve compiled theme files across setup:upgrade calls.
+mkdir /tmp/assets
+cp -pR /app/pub/static/frontend/ /tmp/assets
```

Comment:
If you manually precise the `cd /app` before, you can remove the `/app` in here then.